doc/reference/param.md

```diff
@@ -0,0 +1,163 @@
+# Param API reference
+
+## Main objects
+
+```{eval-rst}
+.. currentmodule:: param
+```
+
+```{eval-rst}
+.. autosummary::
+  :toctree: generated/
+  :nosignatures:
+
+  Parameterized
+  ParameterizedFunction
+  ParamOverrides
+```
+
+## `.param` namespace
+
+These methods and properties are available under the `.param` namespace
+of {py:class}`Parameterized` classes and instances.
+
+```{eval-rst}
+.. currentmodule:: param.parameterized
+```
+
+```{eval-rst}
+.. autosummary::
+  :toctree: generated/
+
+  ~Parameters.add_parameter
+  ~Parameters.debug
+  ~Parameters.defaults
+  ~Parameters.deserialize_parameters
+  ~Parameters.deserialize_value
+  ~Parameters.force_new_dynamic_value
+  ~Parameters.get_param_values
+  ~Parameters.get_value_generator
+  ~Parameters.inspect_value
+  ~Parameters.log
+  ~Parameters.message
+  ~Parameters.method_dependencies
+  ~Parameters.objects
+  ~Parameters.outputs
+  ~Parameters.params
+  ~Parameters.params_depended_on
+  ~Parameters.pprint
+  ~Parameters.print_param_defaults
+  ~Parameters.print_param_values
+  ~Parameters.schema
+  ~Parameters.set_default
+  ~Parameters.set_dynamic_time_fn
+  ~Parameters.set_param
+  ~Parameters.serialize_parameters
+  ~Parameters.serialize_value
+  ~Parameters.trigger
+  ~Parameters.unwatch
+  ~Parameters.update
+  ~Parameters.values
+  ~Parameters.verbose
+  ~Parameters.warning
+  ~Parameters.watch
+  ~Parameters.watch_values
+  ~Parameters.watchers
+```
+
+## Parameterized helpers
+
+```{eval-rst}
+.. currentmodule:: param
+```
+
+```{eval-rst}
+.. autosummary::
+  :toctree: generated/
+
+  param.parameterized.Event
+  param.parameterized.Watcher
+  batch_watch
+  param.parameterized.batch_call_watchers
+  concrete_descendents
+  depends
+  discard_events
+  edit_constant
+  output
+  script_repr
+```
+
+## Logging
+
+```{eval-rst}
+.. autosummary::
+  :toctree: generated/
+
+  param.parameterized.get_logger
+  param.parameterized.logging_level
+```
+
+## Parameters
```

Comment:
Yes true. Though because the Parameters have pretty bare docstrings and Jim wrote https://param.holoviz.org/user_guide/Parameter_Types.html as a reference guide it isn't going to be clear what the reference source is going to be for the Parameters even with an update API page.

Reply:
I think we should just stick to the idea that technical reference material should be found in API reference... and someday maybe the user guide pages will become more how-to-like so there's a more clear distinction